vault/identity_store_aliases.go

```diff
@@ -256,8 +258,46 @@ func (i *IdentityStore) handleAliasUpdateCommon(req *logical.Request, d *framewo
 		return nil, err
 	}
 
+	var existingEntity *identity.Entity
+	if !newAlias {
+		// Verify that the combination of alias name and mount is not
+		// already tied to a different alias
+		if aliasByFactors != nil && aliasByFactors.ID != alias.ID {
+			return logical.ErrorResponse("combination of mount and alias name is already in use"), nil
+		}
+
+		// Fetch the entity to which the alias is tied to
+		existingEntity, err = i.MemDBEntityByAliasID(alias.ID, true)
+		if err != nil {
+			return nil, err
+		}
+
+		if existingEntity == nil {
+			return nil, fmt.Errorf("alias is not associated with an entity")
+		}
+		lockKeys = append(lockKeys, existingEntity.ID)
```

Comment:
Is it possible that `canonical_id` and `existingEntity.ID` can be the same ID, i think they can if you're updating the alias and keeping the same entityID. If so i think the below lock acquisition will deadlock

Reply:
Actually it looks like the LocksForKeys function will dedup them for you